packages/web-ui/src/sections/Document/Editor/Playground/Chat.tsx

```diff
@@ -0,0 +1,217 @@
+import { useCallback, useEffect, useMemo, useRef, useState } from 'react'
+
+import { faker } from '@faker-js/faker'
+import {
+  Chain,
+  CompileError,
+  ContentType,
+  Conversation,
+  Message as ConversationMessage,
+  ConversationMetadata,
+  MessageRole,
+} from '@latitude-data/compiler'
+import { AutoScrollableContainer, Text } from '$ui/ds/atoms'
+import { ErrorMessage, Message, MessageList } from '$ui/ds/molecules'
+import { ChatTextArea } from '$ui/ds/molecules/Chat/ChatTextArea'
+import { cn } from '$ui/lib/utils'
+
+async function* mockAIResponse(
+  response: string,
+  delay = 25,
+): AsyncGenerator<[string, number]> {
+  for (let i = 0; i < response.length; i++) {
+    const minDelay = Math.max(0, delay * 0.5)
+    const maxDelay = Math.max(0, delay * 2)
+    const delayMs = Math.floor(Math.random() * (maxDelay - minDelay) + minDelay)
+    await new Promise((resolve) => setTimeout(resolve, delayMs))
+    yield [response[i]!, 1]
+  }
+}
```

Comment:
AI response is being mocked. There is no AI whatsoever in this PR

Reply:
good approach 👌🏼